packages/forms/test/template_integration_spec.ts

```diff
@@ -1879,3 +2185,50 @@ class NgModelChangesForm {
 class NgModelChangeState {
   onNgModelChange = () => {};
 }
+
+@Component({
+  selector: 'ng-model-max',
+  template: `<form><input name="max" type="number" ngModel [max]="max"></form>`
+})
+class NgModelMaxValidator {
+  max!: number;
+}
+
+@Component({
+  selector: 'ng-model-min',
+  template: `<form><input name="min" type="number" ngModel [min]="min"></form>`
+})
+class NgModelMinValidator {
+  min!: number;
+}
+
+@Component({
+  selector: 'ng-model-min-max',
+  template: `
+    <form><input name="min" type="number" ngModel [min]="min" [max]="max"></form>`
+})
+class NgModelMinMaxValidator {
+  min!: number;
+  max!: number;
+}
+
+@Directive({selector: '[myDir]'})
+export class CustomDirective {
+  @Input() min!: number;
+  @Input() max!: number;
+  constructor() {}
+}
+@Component({
```

Comment:
nit: `constructor` is not needed + add a space between Directive and Component:
```suggestion
}

@Component({
```